diff --git a/CMakeLists.txt b/CMakeLists.txt
index f158dd9f..c5873267 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -161,7 +161,7 @@ include(cmake/libwebrtcbuild.cmake)
 target_link_libraries(tg_owt PUBLIC tg_owt::libwebrtcbuild)
 
 if (UNIX AND NOT APPLE)
-    link_x11(tg_owt)
+    #link_x11(tg_owt)
     link_glib(tg_owt)
 endif()
 
@@ -1939,191 +1939,6 @@ PRIVATE
     modules/remote_bitrate_estimator/remote_estimator_proxy.cc
     sdk/media_constraints.cc
 
-    # screen drawer
-    modules/desktop_capture/screen_drawer.cc
-    modules/desktop_capture/screen_drawer.h
-    modules/desktop_capture/screen_drawer_mac.cc
-    modules/desktop_capture/screen_drawer_linux.cc
-    modules/desktop_capture/screen_drawer_lock_posix.cc
-    modules/desktop_capture/screen_drawer_lock_posix.h
-    modules/desktop_capture/screen_drawer_win.cc
-
-    # primitives
-    modules/desktop_capture/desktop_capture_types.h
-    modules/desktop_capture/desktop_frame.cc
-    modules/desktop_capture/desktop_frame.h
-    modules/desktop_capture/desktop_geometry.cc
-    modules/desktop_capture/desktop_geometry.h
-    modules/desktop_capture/desktop_region.cc
-    modules/desktop_capture/desktop_region.h
-    modules/desktop_capture/shared_desktop_frame.cc
-    modules/desktop_capture/shared_desktop_frame.h
-    modules/desktop_capture/shared_memory.cc
-    modules/desktop_capture/shared_memory.h
-
-    # generic
-    modules/desktop_capture/blank_detector_desktop_capturer_wrapper.cc
-    modules/desktop_capture/blank_detector_desktop_capturer_wrapper.h
-    modules/desktop_capture/cropped_desktop_frame.cc
-    modules/desktop_capture/cropped_desktop_frame.h
-    modules/desktop_capture/cropping_window_capturer.cc
-    modules/desktop_capture/cropping_window_capturer.h
-    modules/desktop_capture/cropping_window_capturer_win.cc
-    modules/desktop_capture/desktop_and_cursor_composer.cc
-    modules/desktop_capture/desktop_and_cursor_composer.h
-    modules/desktop_capture/desktop_capture_options.cc
-    modules/desktop_capture/desktop_capture_options.h
-    modules/desktop_capture/desktop_capturer.cc
-    modules/desktop_capture/desktop_capturer.h
-    modules/desktop_capture/desktop_capturer_differ_wrapper.cc
-    modules/desktop_capture/desktop_capturer_differ_wrapper.h
-    modules/desktop_capture/desktop_capturer_wrapper.cc
-    modules/desktop_capture/desktop_capturer_wrapper.h
-    modules/desktop_capture/desktop_frame_generator.cc
-    modules/desktop_capture/desktop_frame_generator.h
-    modules/desktop_capture/desktop_frame_rotation.cc
-    modules/desktop_capture/desktop_frame_rotation.h
-    modules/desktop_capture/desktop_frame_win.cc
-    modules/desktop_capture/desktop_frame_win.h
-    modules/desktop_capture/differ_block.cc
-    modules/desktop_capture/differ_block.h
-    modules/desktop_capture/fake_desktop_capturer.cc
-    modules/desktop_capture/fake_desktop_capturer.h
-    modules/desktop_capture/fallback_desktop_capturer_wrapper.cc
-    modules/desktop_capture/fallback_desktop_capturer_wrapper.h
-    modules/desktop_capture/full_screen_application_handler.cc
-    modules/desktop_capture/full_screen_application_handler.h
-    modules/desktop_capture/full_screen_window_detector.cc
-    modules/desktop_capture/full_screen_window_detector.h
-    modules/desktop_capture/mouse_cursor.cc
-    modules/desktop_capture/mouse_cursor.h
-    modules/desktop_capture/mouse_cursor_monitor.h
-    modules/desktop_capture/mouse_cursor_monitor_linux.cc
-    modules/desktop_capture/mouse_cursor_monitor_mac.mm
-    modules/desktop_capture/mouse_cursor_monitor_win.cc
-    modules/desktop_capture/resolution_tracker.cc
-    modules/desktop_capture/resolution_tracker.h
-    modules/desktop_capture/rgba_color.cc
-    modules/desktop_capture/rgba_color.h
-    modules/desktop_capture/screen_capture_frame_queue.h
-    modules/desktop_capture/screen_capturer_helper.cc
-    modules/desktop_capture/screen_capturer_helper.h
-    modules/desktop_capture/screen_capturer_darwin.mm
-    modules/desktop_capture/screen_capturer_linux.cc
-    modules/desktop_capture/screen_capturer_win.cc
-    modules/desktop_capture/window_capturer_linux.cc
-    modules/desktop_capture/window_capturer_mac.mm
-    modules/desktop_capture/window_capturer_win.cc
-    modules/desktop_capture/window_finder.cc
-    modules/desktop_capture/window_finder.h
-    modules/desktop_capture/window_finder_mac.h
-    modules/desktop_capture/window_finder_mac.mm
-    modules/desktop_capture/window_finder_win.cc
-    modules/desktop_capture/window_finder_win.h
-
-    # stubs for non-supported
-    modules/desktop_capture/mouse_cursor_monitor_null.cc
-    modules/desktop_capture/screen_capturer_null.cc
-    modules/desktop_capture/window_capturer_null.cc
-
-    # linux specific
-    modules/desktop_capture/linux/base_capturer_pipewire.cc
-    modules/desktop_capture/linux/base_capturer_pipewire.h
-    modules/desktop_capture/linux/mouse_cursor_monitor_x11.cc
-    modules/desktop_capture/linux/mouse_cursor_monitor_x11.h
-    modules/desktop_capture/linux/screen_capturer_x11.cc
-    modules/desktop_capture/linux/screen_capturer_x11.h
-    modules/desktop_capture/linux/shared_x_display.cc
-    modules/desktop_capture/linux/shared_x_display.h
-    modules/desktop_capture/linux/window_capturer_x11.cc
-    modules/desktop_capture/linux/window_capturer_x11.h
-    modules/desktop_capture/linux/window_finder_x11.cc
-    modules/desktop_capture/linux/window_finder_x11.h
-    modules/desktop_capture/linux/window_list_utils.cc
-    modules/desktop_capture/linux/window_list_utils.h
-    modules/desktop_capture/linux/x_atom_cache.cc
-    modules/desktop_capture/linux/x_atom_cache.h
-    modules/desktop_capture/linux/x_error_trap.cc
-    modules/desktop_capture/linux/x_error_trap.h
-    modules/desktop_capture/linux/x_server_pixel_buffer.cc
-    modules/desktop_capture/linux/x_server_pixel_buffer.h
-    modules/desktop_capture/linux/x_window_property.cc
-    modules/desktop_capture/linux/x_window_property.h
-
-    # mac specific
-    modules/desktop_capture/mac/desktop_configuration.mm
-    modules/desktop_capture/mac/desktop_frame_cgimage.h
-    modules/desktop_capture/mac/desktop_frame_cgimage.mm
-    modules/desktop_capture/mac/desktop_frame_iosurface.h
-    modules/desktop_capture/mac/desktop_frame_iosurface.mm
-    modules/desktop_capture/mac/desktop_frame_provider.h
-    modules/desktop_capture/mac/desktop_frame_provider.mm
-    modules/desktop_capture/mac/screen_capturer_mac.h
-    modules/desktop_capture/mac/screen_capturer_mac.mm
-
-    modules/desktop_capture/mac/desktop_configuration.h
-    modules/desktop_capture/mac/desktop_configuration_monitor.cc
-    modules/desktop_capture/mac/desktop_configuration_monitor.h
-    modules/desktop_capture/mac/full_screen_mac_application_handler.cc
-    modules/desktop_capture/mac/full_screen_mac_application_handler.h
-    modules/desktop_capture/mac/window_list_utils.cc
-    modules/desktop_capture/mac/window_list_utils.h
-
-    # win specific
-    modules/desktop_capture/win/cursor.cc
-    modules/desktop_capture/win/cursor.h
-    modules/desktop_capture/win/d3d_device.cc
-    modules/desktop_capture/win/d3d_device.h
-    modules/desktop_capture/win/desktop.cc
-    modules/desktop_capture/win/desktop.h
-    modules/desktop_capture/win/desktop_capture_utils.cc
-    modules/desktop_capture/win/desktop_capture_utils.h
-    modules/desktop_capture/win/display_configuration_monitor.cc
-    modules/desktop_capture/win/display_configuration_monitor.h
-    modules/desktop_capture/win/dxgi_adapter_duplicator.cc
-    modules/desktop_capture/win/dxgi_adapter_duplicator.h
-    modules/desktop_capture/win/dxgi_context.cc
-    modules/desktop_capture/win/dxgi_context.h
-    modules/desktop_capture/win/dxgi_duplicator_controller.cc
-    modules/desktop_capture/win/dxgi_duplicator_controller.h
-    modules/desktop_capture/win/dxgi_frame.cc
-    modules/desktop_capture/win/dxgi_frame.h
-    modules/desktop_capture/win/dxgi_output_duplicator.cc
-    modules/desktop_capture/win/dxgi_output_duplicator.h
-    modules/desktop_capture/win/dxgi_texture.cc
-    modules/desktop_capture/win/dxgi_texture.h
-    modules/desktop_capture/win/dxgi_texture_mapping.cc
-    modules/desktop_capture/win/dxgi_texture_mapping.h
-    modules/desktop_capture/win/dxgi_texture_staging.cc
-    modules/desktop_capture/win/dxgi_texture_staging.h
-    modules/desktop_capture/win/full_screen_win_application_handler.cc
-    modules/desktop_capture/win/full_screen_win_application_handler.h
-    modules/desktop_capture/win/scoped_gdi_object.h
-    modules/desktop_capture/win/scoped_thread_desktop.cc
-    modules/desktop_capture/win/scoped_thread_desktop.h
-    modules/desktop_capture/win/screen_capture_utils.cc
-    modules/desktop_capture/win/screen_capture_utils.h
-    modules/desktop_capture/win/screen_capturer_win_directx.cc
-    modules/desktop_capture/win/screen_capturer_win_directx.h
-    modules/desktop_capture/win/screen_capturer_win_gdi.cc
-    modules/desktop_capture/win/screen_capturer_win_gdi.h
-    modules/desktop_capture/win/screen_capturer_win_magnifier.cc
-    modules/desktop_capture/win/screen_capturer_win_magnifier.h
-    modules/desktop_capture/win/selected_window_context.cc
-    modules/desktop_capture/win/selected_window_context.h
-    modules/desktop_capture/win/wgc_capture_session.cc
-    modules/desktop_capture/win/wgc_capture_session.h
-    modules/desktop_capture/win/wgc_capture_source.cc
-    modules/desktop_capture/win/wgc_capture_source.h
-    modules/desktop_capture/win/wgc_capturer_win.cc
-    modules/desktop_capture/win/wgc_capturer_win.h
-    modules/desktop_capture/win/wgc_desktop_frame.cc
-    modules/desktop_capture/win/wgc_desktop_frame.h
-    modules/desktop_capture/win/window_capture_utils.cc
-    modules/desktop_capture/win/window_capture_utils.h
-    modules/desktop_capture/win/window_capturer_win_gdi.cc
-    modules/desktop_capture/win/window_capturer_win_gdi.h
-
     stats/rtc_stats_report.cc
     stats/rtc_stats.cc
     stats/rtcstats_objects.cc
@@ -2141,10 +1956,6 @@ if (is_x86 OR is_x64)
         common_audio/fir_filter_avx2.cc
         common_audio/fir_filter_avx2.h
     )
-    add_sublibrary(sse2
-        modules/desktop_capture/differ_vector_sse2.cc
-        modules/desktop_capture/differ_vector_sse2.h
-    )
 endif()
 
 if (NOT TG_OWT_USE_PROTOBUF)
diff --git a/cmake/external.cmake b/cmake/external.cmake
index c11df5b6..f3be950c 100644
--- a/cmake/external.cmake
+++ b/cmake/external.cmake
@@ -205,36 +205,6 @@ function(link_glib target_name)
     endif()
 endfunction()
 
-# x11
-function(link_x11 target_name)
-    if (TG_OWT_PACKAGED_BUILD)
-        find_package(X11 REQUIRED COMPONENTS Xcomposite Xdamage Xext Xfixes Xrender Xrandr Xtst)
-        target_include_directories(${target_name}
-        PRIVATE
-            ${X11_X11_INCLUDE_PATH}
-            ${X11_Xlib_INCLUDE_PATH}
-            ${X11_Xcomposite_INCLUDE_PATH}
-            ${X11_Xdamage_INCLUDE_PATH}
-            ${X11_Xext_INCLUDE_PATH}
-            ${X11_Xfixes_INCLUDE_PATH}
-            ${X11_Xrender_INCLUDE_PATH}
-            ${X11_Xrandr_INCLUDE_PATH}
-            ${X11_Xtst_INCLUDE_PATH}
-        )
-        target_link_libraries(${target_name}
-        PRIVATE
-            ${X11_X11_LIB}
-            ${X11_Xcomposite_LIB}
-            ${X11_Xdamage_LIB}
-            ${X11_Xext_LIB}
-            ${X11_Xfixes_LIB}
-            ${X11_Xrender_LIB}
-            ${X11_Xrandr_LIB}
-            ${X11_Xtst_LIB}
-        )
-    endif()
-endfunction()
-
 # PipeWire
 function(link_pipewire target_name)
     find_package(PkgConfig REQUIRED)
diff --git a/cmake/libwebrtcbuild.cmake b/cmake/libwebrtcbuild.cmake
index eb71352a..2f476491 100644
--- a/cmake/libwebrtcbuild.cmake
+++ b/cmake/libwebrtcbuild.cmake
@@ -27,13 +27,6 @@ INTERFACE
     BWE_TEST_LOGGING_COMPILE_TIME_ENABLE=0
 )
 
-if (TG_OWT_USE_PIPEWIRE)
-    target_compile_definitions(libwebrtcbuild
-    INTERFACE
-        WEBRTC_USE_PIPEWIRE
-    )
-endif()
-
 if (NOT TG_OWT_BUILD_AUDIO_BACKENDS)
     target_compile_definitions(libwebrtcbuild
     INTERFACE
@@ -64,10 +57,6 @@ else()
             WEBRTC_MAC
         )
     else()
-        target_compile_definitions(libwebrtcbuild
-        INTERFACE
-            WEBRTC_USE_X11
-        )
     endif()
 
     if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
diff --git a/src/modules/BUILD.gn b/src/modules/BUILD.gn
index 54dffe0a..099d57e9 100644
--- a/src/modules/BUILD.gn
+++ b/src/modules/BUILD.gn
@@ -24,9 +24,6 @@ group("modules") {
     "video_processing",
   ]
 
-  if (rtc_desktop_capture_supported) {
-    deps += [ "desktop_capture" ]
-  }
 }
 
 rtc_source_set("module_api_public") {
@@ -75,10 +72,6 @@ if (rtc_include_tests && !build_with_chromium) {
       "//testing/gtest",
     ]
 
-    if (rtc_desktop_capture_supported) {
-      deps += [ "desktop_capture:desktop_capture_modules_tests" ]
-    }
-
     data = modules_tests_resources
 
     if (is_android) {
@@ -224,10 +217,6 @@ if (rtc_include_tests && !build_with_chromium) {
       "video_processing:video_processing_unittests",
     ]
 
-    if (rtc_desktop_capture_supported) {
-      deps += [ "desktop_capture:desktop_capture_unittests" ]
-    }
-
     data = modules_unittests_resources
 
     if (is_android) {
